src/message/handler/EulerMessageHandler.ts

```
@@ -0,0 +1,73 @@
+import AbstractMessageHandler from './AbstractMessageHandler';
+import { Client, Message } from 'discord.js';
+import { commandPrefix } from '../../constants';
+import { get } from 'https';
+import * as h2p from 'html2plaintext';
+import * as cheerio from 'cheerio';
+import Lang from '../../utils/Lang';
+
+export default class EulerMessageHandler extends AbstractMessageHandler {
+
+    private static readonly MESSAGE_REGEX: RegExp = new RegExp(
+        commandPrefix + 'euler ([0-9]+)( [a-z?]+)?',
+        'gim',
+    );
+
+    public static readonly HANDLER_ID: String = 'euler';
+
+    static canHandle(message: string): boolean {
+        return this.MESSAGE_REGEX.test(message);
+    }
+
+    handle(message: Message, client: Client) {
+        const command = new RegExp(EulerMessageHandler.MESSAGE_REGEX).exec(message.content);
+        const problem = parseInt(command[1], 10);
+        const url = `https://projecteuler.net/problem=${problem}`;
+        const execCommand = command[2];
+
+        if (execCommand === undefined) {
+            message.reply(url);
+            return;
+        }
+
+        if (execCommand.trim() === '?') {
+            this.fetchText(url).then((data) => {
+                const $ = cheerio.load(data);
+                message.reply(
+                    `**Problem ${problem}**`
+                    + '\n```' + h2p($('#content .problem_content').html())
+                    + '\n```',
+                );
+            }).catch((err) => {
+                if (err === 'NOT_FOUND') {
+                    const lang = new Lang('message\\handler\\EulerMessageHandler');
+                    lang.get('err', { problem }).then((data) => {
+                        message.reply(data);
+                    });
+                }
```

this should be logged anywhere. This is actually the beauty of rust. It forces you to not swallow any errors(although it lets you do it if you really/explicitly say it somewhere).

---

Logging error might be another thing: https://github.com/UDCBOT/bot/issues/62 
So lets postpone it

---

`console.error` and a todo-comment is enough for now imo.